gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BomConflictCheckTask.java

```diff
@@ -100,12 +101,15 @@ public final void checkBomConflict() {
                 .collect(Collectors.toMap(
                         Pair::getLeft,
                         Pair::getRight,
-                        // Resolve conflicts by choosing the entry that exactly matches the artifact
-                        (propName1, propName2) -> Stream.of(propName1, propName2)
-                                .filter(propName -> !propName.contains("\\*"))
-                                .findFirst()
-                                .orElseThrow(() -> new IllegalStateException(
-                                        String.format("Conflicts in versions.props:\n%s\n%s", propName1, propName2)))));
+                        // Resolve conflicts by choosing the longer entry because it is more specific
+                        (propName1, propName2) -> {
+                            if (propName1.equals(propName2)) {
```

Comment:
The idea behind the check is good but since that function is applied 2-by-2 but not exhaustively and the keys are not sorted  it's possible that some duplicates slip through. We can always save the previous stream as a list and apply separately a check for that and convert to a map.

Reply:
I'd rather just remove this check. The dependency recommender plugin throws the versions into a `Properties` so it will just take the last one.